### PR TITLE
Add some missing attributes and one API for server support

### PR DESCRIPTION
**Pull request by rhc54**
There are a few attributes servers need in order to support
singleton operations as well as user-specified time limits
on job and spawn operations. Current server module functions
do not include the ability for the server library to pass
a local status from a collective operation (i.e., when the
server library aggregates all local participation before
calling the host)

Add a few job-related error constants for clear reporting
to the user.

Add a useful variant of a data management function.

Signed-off-by: Ralph Castain <rhc@pmix.org>

### CODE REVIEW

**Comment by rhc54** [comment]
Please use emoji reactions ON THIS COMMENT to indicate your position on this proposal.

    You do not need to vote on every proposal
    If you have no opinion, don't vote - that is also useful data
    If you've already commented on this issue, please still vote so
    we know your current thoughts
    Not all proposals solve exactly the same problem, so we may end
    up accepting proposals that appear to have some overlap
    This is not a binding majority-rule vote, but it will be a very
    significant input into the corresponding ASC decision.

Here are the meanings for the emojis:

    Hooray or Rocket: I support this so strongly that I
    want to be an advocate for it
    Heart: I think this is an ideal solution
    Thumbs up: I'd be happy with this solution
    Confused: I'd rather we not do this, but I can tolerate it
    Thumbs down: I'd be actively unhappy, and may even consider
    other technologies instead
    If you want to explain in more detail, feel free to add another
    comment, but please also vote on this comment.

File: Chap_API_Proc_Mgmt.tex [hunk 1]
```diff
@@ -536,6 +554,22 @@ \subsection{Spawn attributes}
...
 Do not generate an event when this job normally terminates.
 }
+%
+\declareAttributeNEW{PMIX_ENVARS_HARVESTED}{"pmix.evar.hvstd"}{bool}{
+Environmental parameters have been harvested by the spawn requestor - the server
+does not need to harvest them.
+}
+%
+\declareAttributeNEW{PMIX_JOB_TIMEOUT}{"pmix.job.time"}{int}{
+Time in seconds before the spawned job should time out and be terminated (0 => infinite)
```
Comment (jjhursey) [inline Chap_API_Proc_Mgmt.tex:564]: Is this a total runtime timeout? Kind of like the walltime limits on batch jobs?

File: Chap_API_Sync_Access.tex [hunk 2]
```diff
@@ -165,6 +165,10 @@ \subsection{Fence-related attributes}
 available to each participant at the end of the operation. By default, this will include all job-level information that was locally generated by \ac{PMIx} servers unless excluded using the \refattr{PMIX_COLLECT_GENERATED_JOB_INFO} attribute.
 }
 %
+\declareAttributeNEW{PMIX_LOCAL_COLLECTIVE_STATUS}{"pmix.loc.col.st"}{pmix_status_t}{
+Status code for local collective operation being reported to the host by the server library
```
Comment (jjhursey) [inline Chap_API_Sync_Access.tex:169]: I had to look back at the PR comment to understand the use case for this
 > e.g., when the server library aggregates all local participation before calling the host

Can we add such a quick note example to the definition here for clarity?

So this is a status value indicating progress in the collective or a status value indicating if the collective completed successfully/erroneously? If the former, then what status values might it send up?

I'm just trying to better understand the expected usage of this attribute.

File: Chap_API_Proc_Mgmt.tex (hunk 1 above)
Comment (rhc54) [reply Chap_API_Proc_Mgmt.tex]: I addressed this - please see if the revised text helps.

File: Chap_API_Sync_Access.tex (hunk 2 above)
Comment (rhc54) [reply Chap_API_Sync_Access.tex]: I addressed this - please see if the revised text helps. Short answer is that it is the latter - a single upcall reporting completion status.

File: Chap_API_Proc_Mgmt.tex (hunk 1 above)
Comment (jjhursey) [reply Chap_API_Proc_Mgmt.tex]: That looks good. Thanks!

File: Chap_API_Sync_Access.tex (hunk 2 above)
Comment (jjhursey) [reply Chap_API_Sync_Access.tex]: That helps. Thanks

**Comment by jjhursey** [comment]
PMIx ASC 1Q 2022

 * Passed the first vote: 11 yes / 0 no / 0 abstain
 * This is can be merged into main/v5 and v4 branches once conflicts are addressed (I'll work on that)

**Review by jjhursey — DISMISSED** [review]

**Comment by jjhursey** [comment]
**Editorial**: Rebased the reference branch without text changes to prepare to merge.

**Review by jjhursey — APPROVED** [review]

**Pull request merged**